gemp-lotr/gemp-lotr-async/src/main/web/src/updated/jCard/types/card-metadata.factory.ts

```diff
@@ -0,0 +1,37 @@
+import { getImageUrl } from "../modules/image-resolution/image-resolver";
+import { isCardASite } from "../modules/sites/types/site.functions";
+import { getBlueprintByCardId } from "./card-formatting.functions";
+import { CardMetadata } from "./card-metadata.interface";
+import { isCardFoil, isCardTengwar } from "./card-options.functions";
+import { buildErrataUrl } from "./errata/errata.functions";
+import { getWikiLink } from "./wiki/wiki.functions";
+
+export const buildCardMetadata = (blueprintId: string): CardMetadata => {
```

Comment:
remember _waaaaaaaaay_ back in jCard.js when I said "it's basically just a ton of logic about building metadata" well this is the updated version of it!

We farm all the business logic out to specific handlers and test them in isolation. We can also write some tests against this and they'll be able to be very naieve because the funky is handled by other people who know more about it

Reply:
and to that point I did not write those tests. Do that.